### PR TITLE
PHP8 Support

### PR DESCRIPTION
**Pull request by javabudd**
References https://github.com/laminas-api-tools/api-tools-api-problem/issues/12 and targets the 1.4.x branch.

Issues with required libraries:
- laminas/laminas-json (https://github.com/laminas/laminas-json/issues/9)
- laminas/laminas-http (https://github.com/laminas/laminas-http/issues/44)
- laminas/laminas-mvc (https://github.com/laminas/laminas-mvc/issues/66)
- laminas/laminas-view awaiting 2.12.x release

### CODE REVIEW

File: src/ApiProblem.php [hunk 1]
```diff
@@ -318,41 +314,33 @@ protected function createDetailFromException()
...
         return $message;
     }
 
     /**
-     * Create HTTP status from an exception.
-     *
      * @return int
      */
-    protected function createStatusFromException()
+    protected function createStatusFromException(): int
```
Comment (javabudd) [inline src/ApiProblem.php:340]: I believe this return type should be reverted

File: src/ApiProblem.php [hunk 2]
```diff
@@ -173,16 +171,14 @@ public function __construct($status, $detail, $type = null, $title = null, array
...
      * @return mixed
      * @throws InvalidArgumentException
      */
-    public function __get($name)
+    public function __get(string $name)
     {
-        $normalized = strtolower($name);
-        if (in_array($normalized, array_keys($this->normalizedProperties))) {
+        $normalized = \strtolower($name);
+        if (\array_key_exists($normalized, $this->normalizedProperties)) {
```
Comment (javabudd) [inline src/ApiProblem.php:181]: logic change

File: src/ApiProblemResponse.php [hunk 3]
```diff
@@ -91,10 +85,6 @@ public function getReasonPhrase()
             return $this->reasonPhrase;
         }
 
-        if (isset($this->recommendedReasonPhrases[$this->statusCode])) {
-            return $this->recommendedReasonPhrases[$this->statusCode];
-        }
-
-        return 'Unknown Error';
+        return $this->recommendedReasonPhrases[$this->statusCode] ?? 'Unknown Error';
```
Comment (javabudd) [inline src/ApiProblemResponse.php:88]: logic change

File: src/Listener/ApiProblemListener.php [hunk 4]
```diff
@@ -69,12 +70,15 @@ public function attach(EventManagerInterface $events, $priority = 1)
...
 
         $sharedEvents = $events->getSharedManager();
-        $sharedEvents->attach(
-            'Laminas\Stdlib\DispatchableInterface',
-            MvcEvent::EVENT_DISPATCH,
-            [$this, 'onDispatch'],
-            100
-        );
+
+        if ($sharedEvents) {
```
Comment (javabudd) [inline src/Listener/ApiProblemListener.php:74]: ensures shared event manager is setup before calling attach on it

File: src/Listener/ApiProblemListener.php [hunk 5]
```diff
@@ -220,7 +224,7 @@ protected function matchAcceptCriteria(AcceptHeader $accept)
     {
         foreach ($this->acceptFilters as $type) {
             $match = $accept->match($type);
-            if ($match && $match->getTypeString() != '*/*') {
+            if ($match && $match->getTypeString() !== '*/*') {
```
Comment (javabudd) [inline src/Listener/ApiProblemListener.php:227]: logic change

File: src/ApiProblem.php [hunk 6]
```diff
@@ -196,27 +192,33 @@ public function __get($name)
...
+        );
     }
 
-    /**
-     * Cast to an array.
-     *
-     * @return array
-     */
-    public function toArray()
+    public function __set(string $name, $value)
```
Comment (javabudd) [inline src/ApiProblem.php:203]: __set and __isset satisfy magic requirements

File: src/ApiProblem.php [hunk 7]
```diff
@@ -243,7 +245,7 @@ public function setDetailIncludesStackTrace($flag)
      */
     protected function getDetail()
     {
-        if ($this->detail instanceof Throwable || $this->detail instanceof Exception) {
+        if ($this->detail instanceof Throwable) {
```
Comment (javabudd) [inline src/ApiProblem.php:248]: since we support only 7.3 and above, remove redundant instanceof Exception conditions

**Comment by weierophinney** [comment]
Closing in favor of #15

**Pull request closed without merging**